From 2fc5832ba5eb3cda2e537329a3c84f6b5410c06f Mon Sep 17 00:00:00 2001
From: Yann Dirson <yann.dirson@vates.tech>
Date: Thu, 11 Jul 2024 11:39:11 +0200
Subject: [PATCH] More py3 fixes

Those were indeed spotted by 2to3 but apparently ignored.

* dict.keys() does return a list any more, and those results cannot be
  concatenated with +
* map() now returns an iterable, which then cannot be used twice. By luck
  it would not be consumed by the debug log, which would issue traces like
  `New lastboot data=<map object at 0x...>`, but then whether it works
  correctly would depend on how code in xcp.net.ifrename evaluates it.


Signed-off-by: Yann Dirson <yann.dirson@vates.tech>
---
 interface-rename.py | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/interface-rename.py b/interface-rename.py
index dfe188f..59ebe0f 100755
--- a/interface-rename.py
+++ b/interface-rename.py
@@ -108,7 +108,7 @@ def run(dryrun, update, args):
             LOG.error("No valid update rules after processing.  Doing nothing")
             return
 
-        all_srule_eths = sr.formulae.keys() + ur.formulae.keys()
+        all_srule_eths = list(sr.formulae.keys()) + list(ur.formulae.keys())
         all_srules = sr.rules + ur.rules
         if ( len(all_srule_eths) != len(set(all_srule_eths)) or
              len(all_srules) != len(set(all_srules)) ):
@@ -152,10 +152,9 @@ def run(dryrun, update, args):
     def macpci_as_list(x):
         return [str(x.mac), str(x.pci), x.tname]
 
-    new_lastboot = map(macpci_as_list, current_state)
+    new_lastboot = [macpci_as_list(x) for x in current_state]
     new_macs = frozenset( (x.mac for x in current_state) )
-    new_old = map(macpci_as_list, filter( lambda x: x.mac not in new_macs,
-                                          dr.lastboot + dr.old ))
+    new_old = [macpci_as_list(x) for x in dr.lastboot + dr.old if x.mac not in new_macs]
 
     LOG.debug("New lastboot data=\n%s" % (niceformat(new_lastboot),))
     LOG.debug("New old data=\n%s" % (niceformat(new_old),))
-- 
2.39.2

